cmd/server/main.go

```diff
@@ -1,11 +1,26 @@
 package main
 
-import (
-	"log"
-)
-
 func main() {
-	if err := run(); err != nil {
-		log.Fatalf("Application terminated due to a critical error: %v", err)
+	mainCtx, mainCtxCancel := mainContext()
+	defer mainCtxCancel()
+
+	logger, loggerSync := loggerWithSyncFunc()
```

Comment:
Почему не просто

```suggestion
	logger := loggerWithSyncFunc()
    defer logger.Sync()
```

?

Reply:
✅ Исправлено в коммите: [2525575](https://github.com/gdyunin/metricol/pull/20/commits/2525575028e9c16c334918bb0a4df53448f2b628)

Reply:
А было скорее из перфекционизма. Вот к чему адресован перфекционизм.

Выше есть код:
```go
mainCtx, mainCtxCancel := mainContext()
defer mainCtxCancel()
```
Который псевдокодом выглядит так:
```go
сущность, закрывающаяФункция = генерироватьСущность()
defer закрывающаяФункция()
```

Хотелось и с логгером к этому же шаблону присоединиться. Сейчас посмотрел и понял, что это избыточно, особенно в `main.go`